irohad/cli/flags.hpp

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_FLAGS_HPP_
+#define IROHA_FLAGS_HPP_
+
+#include <CLI/CLI.hpp>
+#include <boost/assert.hpp>
+#include "ametsuchi/config.hpp"
+#include "common.hpp"
+#include "env-vars.hpp"
+#include "torii/config.hpp"
+#include "util/network.hpp"
+
+namespace flag {
+
+  using namespace std::literals::string_literals;
+
+  enum { IROHA_PORT_MIN = 1, IROHA_PORT_MAX = 65535 };
+  enum { IROHA_DELAY_MIN = 1, IROHA_DELAY_MAX = 100000 };
+  enum { IROHA_SIZE_MIN = 1, IROHA_SIZE_MAX = 100000 };
+
+  inline void addPeerFlags(CLI::App *p,
+                           iroha::config::Peer *peer,
+                           iroha::torii::config::Torii *torii,
+                           iroha::config::Cryptography *crypto) {
+    BOOST_ASSERT(p);
+    BOOST_ASSERT(torii);
+    BOOST_ASSERT(crypto);
+    BOOST_ASSERT(peer);
+
+    // both flags can be used for the same thing
+    p->add_option("--api-host,--torii-host"s,        /* option's name */
+                  torii->host,                       /* bind to this variable */
+                  "Client API (torii) listen host"s, /* description */
+                  true /* use initial value as default */)
+        ->envname(IROHA_TORII_HOST)
+        ->group("Peer"s)
+        ->check(iroha::network::is_host_valid);
+
+    p->add_option("--api-port,--torii-port"s,
+                  torii->port,
+                  "Client API (torii) listen port"s,
+                  true)
+        ->envname(IROHA_TORII_PORT)
+        ->check(CLI::Range(IROHA_PORT_MIN, IROHA_PORT_MAX))
+        ->group("Peer"s);
+
+    p->add_option("--host"s, peer->host, "Peer's listen address"s, true)
+        ->envname(IROHA_TORII_HOST)
+        ->group("Peer"s)
+        ->check(iroha::network::is_host_valid);
+
+    p->add_option("--port"s, peer->port, "Peer's listen port"s, true)
+        ->envname(IROHA_TORII_PORT)
+        ->check(CLI::Range(IROHA_PORT_MIN, IROHA_PORT_MAX))
+        ->group("Peer"s);
+
+    p->add_option(
+         "--pubkey"s, crypto->public_key, "Path to peer's public key"s, false)
+        ->required()
+        ->envname(IROHA_PEER_PUBKEY)
```

Comment:
Should be converted to absolute path in advance, same for private key

Reply:
Good point. I will work on this.